src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,49 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  deleteAnimal() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
The deleteAnimal method currently reassigns the Animal.alive array each time an animal's health drops below zero. Consider using a more efficient way to remove dead animals without having to recreate the array each time, such as directly splicing the dead animal out of the array.